apps/backend/src/modules/catalog/cache-refresh.ts

```diff
@@ -0,0 +1,185 @@
+  const script = `
+    if redis.call("get", KEYS[1]) == ARGV[1] then
+      return redis.call("del", KEYS[1])
+    else
+      return 0
+    end`;
+
+  try {
+    await redis.eval(script, {
+      keys: [LOCK_KEY],
+      arguments: [INSTANCE_ID],
+    });
+  } catch (error) {
+    console.warn(
+      `[CatalogCache][${INSTANCE_ID}] Failed to release refresh lock: ${String(
+        error
+      )}`
+    );
+  }
+};
+
+const fetchActiveTerms = async (): Promise<ParsedTerm[]> => {
+  const rawTerms = await TermModel.find({
+    academicCareerCode: "UGRD",
+    temporalPosition: { $in: ["Current", "Future"] },
+  })
+    .select({ name: 1 })
+    .lean();
+
+  const seen = new Map<string, ParsedTerm>();
+  for (const term of rawTerms) {
+    if (!term?.name) continue;
+
+    const parsed = parseTermName(term.name);
+    if (!parsed) continue;
+
+    const key = `${parsed.year}-${parsed.semester}`;
+    if (!seen.has(key)) {
+      seen.set(key, parsed);
+    }
+  }
+
+  return Array.from(seen.values());
+};
+
+const warmTermCacheForCatalogQuery = async (
+  redis: RedisClientType,
+  term: ParsedTerm
+): Promise<void> => {
+  const { year, semester } = term;
+  const includesGradeDistribution = true; // matches GetCatalog usage
+
+  try {
+    const classes = await fetchCatalogData(
+      year,
+      semester,
+      includesGradeDistribution
+    );
+
+    const cacheKey = buildCatalogCacheKey(
+      year,
+      semester,
+      includesGradeDistribution
+    );
+
+    await storeCatalogInCache(redis, cacheKey, classes);
+
+    console.info(
+      `[CatalogCache][${INSTANCE_ID}] Warmed with-grade cache for ${semester} ${year}`
+    );
+  } catch (error) {
+    console.warn(
+      `[CatalogCache][${INSTANCE_ID}] Failed to warm with-grade cache for ${semester} ${year}: ${String(
+        error
+      )}`
+    );
+  }
+};
+
+const runRefresh = async (redis: RedisClientType): Promise<void> => {
+  const acquired = await acquireLock(redis);
+  if (!acquired) {
+    return;
+  }
+
+  console.info(`[CatalogCache][${INSTANCE_ID}] Acquired refresh lock`);
+
+  const start = Date.now();
+
+  try {
+    const terms = await fetchActiveTerms();
+    if (terms.length === 0) {
+      console.info(
+        `[CatalogCache][${INSTANCE_ID}] No active terms found for catalog cache refresh`
+      );
+      return;
+    }
+
+    for (const term of terms) {
+      await warmTermCacheForCatalogQuery(redis, term);
+    }
+
+    const elapsedMs = Date.now() - start;
+    console.info(
+      `[CatalogCache][${INSTANCE_ID}] Completed catalog cache refresh in ${elapsedMs}ms`
+    );
+  } catch (error) {
+    console.warn(
+      `[CatalogCache][${INSTANCE_ID}] Catalog cache refresh errored: ${String(
+        error
+      )}`
+    );
+  } finally {
+    await releaseLock(redis);
+  }
+};
+
+export const registerCatalogCacheRefresh = (redis: RedisClientType): void => {
+  if (process.env.DISABLE_CATALOG_CACHE_REFRESH === "true") {
```

Comment:
This env var should be added to `src/config.ts`. 

When would this variable be true?